kong/utils.go

```diff
@@ -103,3 +111,30 @@ func HTTPClientWithHeaders(client *http.Client,
 	}
 	return res
 }
+
+//ParseSemanticVersion creates a semantic version from a string
+// fix enterprise edition semver adding patch number
+// fix enterprise edition version with dash
+// fix bad version formats like 0.13.0preview1
+func ParseSemanticVersion(v string) (semver.Version, error) {
+	re := regexp.MustCompile(`(\d+\.\d+)(?:[\.-](\d+))?(?:\-?(.+)$|$)`)
+	m := re.FindStringSubmatch(v)
+	if len(m) != versionParts {
+		return semver.Version{}, fmt.Errorf("Unknown Kong version : '%v'", v)
+	}
+	if m[2] == "" {
+		m[2] = "0"
+	}
+	if m[3] != "" {
+		m[3] = "-" + strings.Replace(m[3], "enterprise-edition", "enterprise", 1)
+		m[3] = strings.Replace(m[3], ".", "", -1)
+	}
+	v = fmt.Sprintf("%s.%s%s", m[1], m[2], m[3])
+	return semver.Make(v)
+}
+
+//VersionFromInfo retrieves the version from the response of root
+// or /kong endpoints
+func VersionFromInfo(info map[string]interface{}) string {
+	return info["version"].(string)
```

Comment:
This can panic if `version` doesn't exist in info.
Please use `v, ok := myMap["key"]` check here.